web-registry/src/pages/CreditClassDetailsWithContent.tsx

```diff
@@ -0,0 +1,322 @@
+      <></>
+    );
+  };
+
+  const onCtaClick = (): void => {
+    if (isLandSteward && content) {
+      return onBtnClick(openModalLink, content?.landSteward?.ctaButton);
+    } else {
+      return setBuyerModalOpen(true);
+    }
+  };
+
+  const openModalLink = (modalLink: string): void => {
+    setModalIframeLink(modalLink);
+  };
+
+  const setContent = (): void => {
+    navigate(
+      `/credit-classes/${creditClassId}/${
+        isLandSteward ? 'buyer' : 'land-steward'
+      }`,
+    );
+  };
+
+  return (
+    <div className={styles.root}>
+      <HeroTitle
+        isBanner
+        img={hero}
+        title={
+          isLandSteward
+            ? content?.landSteward?.heroSection?.title
+            : content?.buyer?.heroSection?.title
+        }
+        descriptionRaw={
+          isLandSteward
+            ? content?.landSteward?.heroSection?.descriptionRaw
+            : content?.buyer?.heroSection?.descriptionRaw
+        }
+        classes={{ main: styles.heroMain }}
+      />
+      <Section
+        title={
+          content?.nameRaw ? <BlockContent content={content?.nameRaw} /> : ''
+        }
+        classes={{
+          root: styles.introSection,
+          title: styles.title,
+          titleWrap: styles.titleWrap,
+        }}
+      >
+        <Description className={styles.sectionDescription}>
+          <BlockContent content={content?.descriptionRaw} />
+        </Description>
+      </Section>
+      <div className="topo-background-alternate">
+        {content?.ecologicalImpact && (
+          <ImpactSection
+            title="Ecological Impact"
+            impacts={content?.ecologicalImpact}
+          />
+        )}
+        {!isLandSteward && (
+          <CreditClassOverviewSection
+            className={styles.overviewSection}
+            creditClass={creditClass}
+            nameRaw={content?.nameRaw}
+            overviewCards={content?.overviewCards}
+            sdgs={content?.sdgs}
+          />
+        )}
+      </div>
+      {isLandSteward && (
+        <>
+          {getFeaturedProjects()}
+          <div className="topo-background-alternate">
+            <CreditClassOverviewSection
+              creditClass={creditClass}
+              nameRaw={content?.nameRaw}
+              overviewCards={content?.overviewCards}
+              sdgs={content?.sdgs}
+            />
+          </div>
+        </>
+      )}
+      {!isLandSteward && getAllProjects()}
+      {isLandSteward &&
+        content?.landSteward?.steps?.map(
+          (
+            stepSequence: {
+              title: Maybe<string> | undefined;
+              preTitle: Maybe<string> | undefined;
+              descriptionRaw: any;
+              stepCards: Maybe<Maybe<StepCardFieldsFragment>[]> | undefined;
+            },
+            index: React.Key | null | undefined,
+          ) => (
+            <div
+              className={cx('topo-background-alternate', styles.flex)}
+              key={index}
+            >
```

Comment:
```suggestion
            <Box key={index} sx={{ display: 'flex' }} className="topo-background-alternate">
```
I mentioned this in [victor's PR](https://github.com/regen-network/regen-web/pull/856#discussion_r819893049), but we should probably avoid using `makeStyles` as much as possible moving forward, given it probably won't work in react 18

Reply:
I'm in agreement, but.... 
This is probably a sign that my code isn't modular enough, but I haven't figured out how to re-use styles with `styled` without a massive refactor of all the components it's touching. 

In this case, I mostly just renamed this file to separate the "content" credit class pages from the "simple," so this part is copy-paste.

In other cases, it is like unscrambling an omelette. But I will try to use `styled` moving forward.

Reply:
copy-paste makes sense! There will definitely be places we need to `makeStyles` to pass some custom css to components which accept `classes` or `className` as props. 

I don't think `styled` is a great replacement for this sort of thing either either... it's good for components moving forward, but requires some different reasoning for passing styles via props. We could also just use `sx` for everything and avoid `styled` for most... there's a small perf hit compared to `styled` or passing a css class, but it's minor.  

I've been playing with some ideas, and it seems that `styled` + `sx` (which you can pass via props and combine) works great, and `makeStyles` + `sx` generally works well (sometimes passed `sx` props will be overridden by the `makeStyles` class), but combining the 3 can get a bit wonky. Sometimes it works perfectly, other times not so much, and i'm not entirely sure why. My current plan is to use `styled` for base styles (anything that won't be modified through different uses cases), `sx` for anything that might change, and handling modifications through a passed `sx` prop as [mui suggests](https://mui.com/system/the-sx-prop/#passing-sx-prop)

That's all a tangent though. TLDR: no worries at all and no need to change anything, just wanted to make sure you're aware. I think everyone is at this point though, so i'll stop reminding everyone 😀

Reply:
I've created a discussion on GitHub to formalize this topic with some priority. I have also dedicated a space to it in our Dev Docs in Notion..

[Styling strategies for MUI v3 to v5](https://github.com/regen-network/regen-web/discussions/863)